### DISCUSSION TITLE
How to find the colour temperature a jpg image was shot at?

### DISCUSSION THREAD

**Discussion by ujjayants** [colour-science/colour#887] [Q&A]
Is it possible to retrieve the temperature at which an image (in jpg format) was shot at using your code? If yes, how? Thanks.

**Comment by thomasmansencal** [comment]
Hi @ujjayants,

Short answer it is impossible! An infinity of lighting conditions, camera sensitivities and image processing pipelines will generate the same image. You can get an estimate by making some assumptions but it is not trivial. It is a good fit task for ML though, look for something along the lines of "learning illuminant estimation from photograph" e.g. https://ieeexplore.ieee.org/document/8451229 There are a lot of traps on the road, for example, what if the white balance selected for processing is not the neutral one for the scene?

Cheers,

Thomas

**Reply by thomasmansencal** [reply]
As @nick-shaw was pointing out, it might also be possible to use EXIF data if still present. I would not rely on that in a generic pipeline that ingests random consumer data though.

**Reply by ujjayants** [reply]
Hi. I tried getting the dominant RGB value from an image using k-means and then using the Hernandez method to get the CCT.
This seems to work with incandescent (less than 3000 K), fluorescent ( 3000-4500 K).
However, it doesn't get good results for images taken outdoors in natural light. Ideally, images taken in daylight/open shade should be returning 5000-6500 K. What I instead got were values in the fluorescent range.

**Reply by thomasmansencal** [reply]
K-Means or anything that "naively" analyses the colour of the images is bound to fail. Imagine someone capturing a patch of sky with a zoom lens, setting the whitebalance at 3200K for one image and 6500K for the second one, how would you find the whitebalance used in that case?
